desc/backend.py

```diff
@@ -134,8 +147,62 @@ def wrapper(*args, **kwargs):
 
         return wrapper
 
-    # JAX implementation is not differentiable on gpu.
-    eigh_tridiagonal = execute_on_cpu(jax.scipy.linalg.eigh_tridiagonal)
+    _eigh_tridiagonal = jnp.vectorize(
+        jax.scipy.linalg.eigh_tridiagonal,
+        signature="(m),(n)->(m)",
+        excluded={"eigvals_only", "select", "select_range", "tol"},
+    )
+    if desc_config["kind"] == "gpu":
+        # JAX eigh_tridiagonal is not differentiable on gpu.
+        # https://github.com/jax-ml/jax/issues/23650
```

Comment:
that issue is still waiting on a MRE, is this still an issue? and in what cases?

Reply:
Is there a case where we are doing eigendecomp on a CPU? We should avoid that, even if it means always calculating both eigenvalues and eigenfunctions.

Reply:
jax.scipy.linalg.eigh calls jax.lax.linalg.eigh which is calculating both eigenvalues and eigenvectors so I guess we are saving some memory by not storing the eigenvector array when we just use compute? So the option of having eigenvectors is only saving memory a few GBs when we compute the eigvals on multiple surfaces.

When we do optimization, I don't think we are saving anything because eigenvectors are needed to calculate the gradient of the eigenvalue.
 
In terms of performance it should be exactly as fast as the previous version.

Reply:
eigenvector matrix is big so if we can save gigabytes with an if statement it's worth it